opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -244,20 +244,63 @@ def force_flush(self, timeout_millis: int = 30000) -> bool:
         return True
 
 
+# skip natural LogRecord attributes
+# http://docs.python.org/library/logging.html#logrecord-attributes
+_RESERVED_ATTRS = frozenset(
+    (
+        "args",
+        "asctime",
+        "created",
+        "exc_info",
+        "exc_text",
+        "filename",
+        "funcName",
+        "levelname",
+        "levelno",
+        "lineno",
+        "module",
+        "msecs",
+        "message",
+        "msg",
+        "name",
+        "pathname",
+        "process",
+        "processName",
+        "relativeCreated",
+        "stack_info",
+        "thread",
+        "threadName",
+    )
+)
+
+
 class OTLPHandler(logging.Handler):
     """A handler class which writes logging records, in OTLP format, to
     a network destination or file.
     """
 
-    def __init__(self, level=logging.NOTSET, log_emitter=None) -> None:
+    def __init__(
+        self,
+        level=logging.NOTSET,
+        log_emitter=None,
+        *,
+        attributes_key: Optional[str] = None,
```

Comment:
We can add an `exclude` parameter (that defaults to and overrides `_RESERVED_ATTRS`) or an `include` parameter that subtracts from `_RESERVED_ATTRS`.

Reply:
Where is this defined in spec? For anything not defined in the spec, I would suggest propose the idea to logs spec. We certainly don't want to add extra capabilities that are not defined in spec to this prototype.

Reply:
The spec says:
- Attributes are optional
- Attributes are of the form `mapping<str, any>`
- We should use the capabilities of existing logging systems

Which is what I am doing here.
Can you please clarify what part you are questioning?

If you are referring to the `attributes_key`, that's a necessary workaround to limitations of Python's logging system: you can't directly set attributes that collide with the existing LogRecord attributes (e.g. `name`). This allows users to work around that. I don't see how an implementation detail of cPython could be part of the specs.

Reply:
I was referring to log attribute filtering. If that is something language sdks need to include because the collector has the processor built for this https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/attributesprocessor.

Reply:
And speaking of `attributes_key`, if it is set no other attributes are considered why? If you want to avoid a collision with existing log attributes you could prefix with something `my_custom_attrs.name` which is the usual pattern in otel.